server/app/forms/QuestionForm.java

```diff
@@ -53,49 +59,64 @@ protected QuestionForm(QuestionDefinition qd) {
     }
   }
 
-  public String getQuestionName() {
+  public final String getQuestionName() {
```

Comment:
question (non-blocking): why set all of these `final`?

Reply:
I set these to final since this is an abstract class and there's no need to allow subclasses to arbitrarily override the implementation. If that need should arise, we can unmark them as final. It's the same logic as marking a class as "final" since the defaults would otherwise leave them "open" to reimplementation.